dask/sharedict.py

```diff
@@ -0,0 +1,96 @@
+from toolz import concat, unique, count
+from collections import Mapping
+
+
+class ShareDict(Mapping):
+    """ A Mapping composed of other Mappings
+
+    This is a union of other disjoint mappings.  It allows the combination of
+    many dicts into a single dict-like object without creating copies of the
+    underlying dicts.  It provides cheap ``update``, ``len`` and ``__iter__``
+    operations as well as a fairly cheap ``__getitem__`` operation (linear in
+    the number of constituent mappings).
+
+    This class is optimized for Dask's use, and may not be generally useful.
+    Users may want to consider the standard ``collections.ChainMap`` data
+    structure.
+
+    This class makes the following assumptions:
+
+    1.  Constituent mappings are disjoint.  No key is in more than one mapping.
```

Comment:
It seems this assumption isn't enforced (at least for `len` and `__iter__`)? Perhaps change the docs to reflect this?